tests/auth/managers/simple/routes/test_login.py

```diff
@@ -0,0 +1,82 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from unittest.mock import patch
+
+import pytest
+
+from tests_common.test_utils.config import conf_vars
+
+TEST_USER_1 = "test1"
+TEST_ROLE_1 = "viewer"
+TEST_USER_2 = "test2"
+TEST_ROLE_2 = "admin"
+
+
+@pytest.mark.db_test
+class TestLogin:
```

Comment:
Ideally, these tests should be in `test_login` in `services`. Because they test the business logic that is contained in `services/login` now. And then in here `router/test_login` we should have 2 tests, one per endpoint, that are mocking `Login`. These tests should just check that the value returned by `Login` is passed through the wire

Reply:
Make sense! Separated the service layer test and route test. Service one calls the service and route one using the `test_client`. I ended up with 4 tests though because mocking is a bit weird due to LoginResponse normally not returning HTTPException but raising them. It should be fine to have them separate